hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/ImmutableSegment.java

```diff
@@ -30,10 +43,66 @@
 @InterfaceAudience.Private
 public class ImmutableSegment extends Segment {
 
+  private boolean isFlat;   // whether it is based on CellFlatMap or ConcurrentSkipListMap
```

Comment:
This is a private knowledge it is not externally observable. 
We have discussed that for a long time and agreed that it doesn't suit CellSet.